buster/busterbot.py

```diff
@@ -90,31 +103,52 @@ def process_input(
         if not user_input.endswith("\n"):
             user_input += "\n"
 
+        user_inputs = UserInputs(original_input=user_input)
+
         # The returned message is either a generic invalid question message or an error handling message
         question_relevant, irrelevant_question_message = self.validator.check_question_relevance(user_input)
 
         if question_relevant:
             # question is relevant, get completor to generate completion
-            matched_documents = self.retriever.retrieve(user_input, sources=sources, top_k=top_k)
+
+            # reformulate the question if a reformulator is defined
+            if self.question_reformulator is not None and reformulate_question:
+                reformulated_input, reformulation_error = self.question_reformulator.reformulate(
+                    user_inputs.original_input
+                )
+                user_inputs.reformulated_input = reformulated_input
+
+                if reformulation_error:
+                    completion = Completion(
+                        error=True,
+                        user_inputs=user_inputs,
+                        matched_documents=pd.DataFrame(),
+                        answer_text="Something went wrong generating the question...",  # TODO: Better error handling?
```

Comment:
I'd be inclined to skip the reformulation in this case and go to the next step of the workflow.

Reply:
my rationale is we should always be "clear" which formulation of the question we are using, so if something goes wrong generating the question we shouldn't continue otherwise it's not clear to the user which prompt is used

Reply:
overall we agree that this would require better overall error handling which is outside the scope of this PR